packages/sx.js/src/clients/offchain/ethereum-sig/types.ts

```diff
@@ -3,28 +3,40 @@ export const domain = {
   version: '0.1.4'
 };
 
-export const SingleChoiceVoteTypes = {
-  Vote: [
-    { name: 'from', type: 'address' },
-    { name: 'space', type: 'string' },
-    { name: 'timestamp', type: 'uint64' },
-    { name: 'proposal', type: 'string' },
-    { name: 'choice', type: 'uint32' },
-    { name: 'reason', type: 'string' },
-    { name: 'app', type: 'string' },
-    { name: 'metadata', type: 'string' }
-  ]
+const baseVoteTypes = [
+  { name: 'from', type: 'address' },
+  { name: 'space', type: 'string' },
+  { name: 'timestamp', type: 'uint64' },
+  { name: 'proposal', type: 'string' },
+  { name: 'reason', type: 'string' },
+  { name: 'app', type: 'string' },
+  { name: 'metadata', type: 'string' }
+];
+
+export const basicVoteTypes = {
+  Vote: [...baseVoteTypes, { name: 'choice', type: 'uint32' }]
+};
+
+export const singleChoiceVoteTypes = basicVoteTypes;
+
+export const approvalVoteTypes = {
+  Vote: [...baseVoteTypes, { name: 'choice', type: 'uint32[]' }]
```

Comment:
Looks like we order is important here and it won't work. When I try to vote it fails with `wrong types` error.

If I move `choice` back to where it was it works. I'd say we just forget `baseVoteTypes` and have it duplicate if needed to make sure voting works.